aten/src/ATen/native/cuda/ScatterGatherKernel.cu

```diff
@@ -415,7 +439,7 @@ struct _cuda_scatter_fill_internal_kernel {
     auto loop = [=]C10_DEVICE(int i) {
       auto offsets = offset_calc.get(i);
 
-      int64_t idx_dim = *(int64_t*)(index_ptr + offsets[1]);
+      int64_t idx_dim = *(index_t*)(index_ptr + offsets[1]);
```

Comment:
@ngimel, the index is cast to `int64_t` here. Is that what you meant?

Reply:
right, that should guarantee correctness